interview-questions.md

```diff
@@ -4,58 +4,91 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. Why would you use the method super()?
 
-  Your answer:
+  Your answer: The method super() is used in order to pass inheritance between parent and child classes.
 
   Researched answer:
 
+Super is used within the constructor method to pass attributes from the constructor in the parent class. Without the super method the attributes would not be inheritable.
+
 
 
 2. What is a virtual DOM?
 
   Your answer:
 
+  All I really knew about the research is that DOM stands for Document Object Model.
+
   Researched answer:
 
+React creates a virtual DOM which allows for quicker manipulation then accessing the real DOM. This is used when a component is rendered and the virtual DOM checks the real DOM. This allows for only parts that require a change to render instead of the whole DOM.
+
 
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
 
   Your answer:
 
+You need to have a render and a return. You must also have a react fragment enclosing code that you want rendered.
+
+
+
   Researched answer:
 
+I need to do more research and have a better understanding of how state is and classes work with react. I had a hard time knowing what to use classes within a react application.
 
 
 4. What is JSX?
 
   Your answer:
 
+  It is a way to be able to write HTML on a javascript file. It allows for both the Javascript and the HTML to interact easily.
+
   Researched answer:
 
+JSX is a syntax extension of regular javascript used to create react elements. This allows for HTML like code and javascript to co-exist.
 
 
 5. What is state in React?
 
   Your answer:
 
+  State is a way to store behaviors or values that belong to a component. It is used in React.
+
   Researched answer:
 
+State is a javascript object that sores a components dynamic data that determines the components behavior. State is used to track changing information between renders which allows for it to be dynamic and interactive. State can only be used within class components
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
   Your answer:
 
+  I only have heard this term get thrown around in class a couple of times but don't really have an idea of what it means.
```

Comment:
Think back to when you had your variables below your function. This means javascript, when you called your function, had to go look somewhere OTHER than above the function for the variable holding the data you needed to make the function work correctly. So it looked, found the variable/data and HOISTED it back up to above the function to make it work. 

This can impact performance, and generally give unexpected results. Its kind of a no-no in the coding world.